apps/aecore/lib/aecore/structures/signed_tx.ex

```diff
@@ -25,13 +26,33 @@ defmodule Aecore.Structures.SignedTx do
   use ExConstructor
 
   @spec is_coinbase?(SignedTx.t()) :: boolean()
-  def is_coinbase?(tx) do
-    tx.data.from_acc == nil && tx.signature == nil
+  def is_coinbase?(%{data: %{from_acc: key}, signature: signature}) do
+    key == nil && signature == nil
   end
 
   @spec is_valid?(SignedTx.t()) :: boolean()
-  def is_valid?(tx) do
-    tx.data.value >= 0 && tx.data.fee >= 0 && Keys.verify_tx(tx)
+  def is_valid?(%{data: data, signature: signature}) do
+    data.value >= 0 && data.fee >= 0 &&
+      Signing.verify(Serialization.pack_binary(data), signature, data.from_acc)
+  end
+
+  @doc """
+  Takes the transaction that needs to be signed
+  and the private key of the sender.
+  Returns a signed tx
+
+  ## Parameters
+     - tx: The transaction data that it's going to be signed
+     - priv_key: The priv key to sign with
+
+  """
+  @spec sign_tx(SpendTx.t(), binary()) :: {:ok, SignedTx.t()}
+  def sign_tx(%SpendTx{} = tx, priv_key) when byte_size(priv_key) == 32 do
```

Comment:
is it possible to not pass the priv_key here, but have it in the `Signing.sign` method?

Reply:
No, I need the private key to be passed to the `Aewallet.Signing.sing()` function, otherwise it cannot sign the data

Reply:
can you fetch it in `sign_tx`?

Reply:
Well, I guess I could call `Wallet.get_private_key()`

Reply:
ok, would be better than passing the private_key every time

Reply:
can't this be called in the `Signing.sign` function?

Reply:
This is the function in the `elixir-wallet` repo

Reply:
ok, thanks, I didn't notice

Reply:
@gspasov the private-ley is still passed, could you pass `Wallet.get_private_key()` as mentioned?